modules/vault-cluster/main.tf

```diff
@@ -145,19 +145,26 @@ module "security_group_rules" {
 resource "aws_iam_instance_profile" "instance_profile" {
   name_prefix = "${var.cluster_name}"
   path        = "${var.instance_profile_path}"
-  role        = "${aws_iam_role.instance_role.name}"
+  role        = "${data.template_file.instance_role_name.rendered}"
 
   # aws_launch_configuration.launch_configuration in this module sets create_before_destroy to true, which means
   # everything it depends on, including this resource, must set it as well, or you'll get cyclic dependency errors
   # when you try to do a terraform destroy.
   lifecycle {
     create_before_destroy = true
   }
+
+  # We must explictly inform terraform of the following dependencies as terraform is unable to resolve them automatically
+  # with a simple string being passed to this resource's role argument (as opposed to an actaul aws_iam_role resource)
+  depends_on = ["aws_iam_role.prefixed_instance_role", "aws_iam_role.named_instance_role"]
```

Comment:
Hm, the dependency on `data.template_file.instance_role_name` should be enough, since that data source already depends on the two `aws_iam_role` resources.